tensorly/tenalg/einsum_tenalg/generalised_inner_product.py

```diff
@@ -1,11 +1,12 @@
 from ... import backend as T
-import numpy as np
+from .caching import einsum_path_cached
 
 # Author: Jean Kossaifi
 # License: BSD 3 clause
 
 
-def inner(tensor1, tensor2, n_modes=None):
+@einsum_path_cached
+def inner_path(tensor1, tensor2, n_modes=None):
```

Comment:
Will that not set ``key=tensor1``? will that give the correct behaviour for the cache?

Reply:
No the key is only used inside the wrapper to retrieve the cached version but not actually passed to the function. I need to document that clearly.